caching-attribute-service-client/src/main/java/org/hypertrace/core/attribute/service/cachingclient/DefaultCachingAttributeClient.java

```diff
@@ -84,4 +98,25 @@ private NoSuchElementException buildErrorForMissingAttribute(String scope, Strin
     return new NoSuchElementException(
         String.format("No attribute available for scope '%s' and key '%s'", scope, key));
   }
+
+  private NoSuchElementException buildErrorForMissingAttribute(String attributeId) {
+    return new NoSuchElementException(
+        String.format("No attribute available for id '%s'", attributeId));
+  }
+
+  private void loadScopeAndKeyCache(AttributeMetadata attributeMetadata) {
+    this.scopeAndKeyLookup.put(
+        attributeMetadata.getId(),
+        new AttributeScopeAndKey(attributeMetadata.getScope().name(), attributeMetadata.getKey()));
+  }
+
+  private static final class AttributeScopeAndKey {
+    private final String scope;
+    private final String key;
+
+    private AttributeScopeAndKey(String scope, String key) {
```

Comment:
Does this need an implementation of equals and hashCode?

Reply:
Shouldn't, this isn't used as a key anywhere, only as a map value

Reply:
ok.